index.js

```diff
@@ -106,13 +120,40 @@ class I18n {
         }
 
         const type = typeof result;
-        result = type === 'string' ? interpolate(result, data) : result;
+        result = type === 'string' ? interpolate(result, params) : result;
 
-        return ACCEPTABLE_RETURN_TYPES.includes(type)
-            ? result
-            : this[MISSING](
-                `${key}`, this.$scope, this.translations
-            ) || I18n.getDefault(...keys);
+        if (ACCEPTABLE_RETURN_TYPES.includes(type)) {
+            if (isTemplateInjectionEligible(result)) {
+                return this.handleTemplateInjection(key, result, templates, templatesTransformer);
+            }
+
+            return result;
+        }
+
+        return this[MISSING](`${key}`, this.$scope, this.translations) || I18n.getDefault(...keys);
+    }
+
+    /**
+     * Inject all templates into the origin translation.
+     *
+     * @param {String} key String representing dot notation
+     * @param {String} originTranslation The translation into which the templates will be injected.
+     * @param {Record.<String, Function>} templates The templates that will be injected.
+     * @param {Record.<String, Function>} templatesTransformer The templates transformer function
+     * @return {String} template injected translation
+     */
+    handleTemplateInjection(key, originTranslation, templates, templatesTransformer) {
+        try {
+            return injectTemplates({
+                originTranslation,
+                templates,
+                templatesTransformer
+            });
+        } catch (error) {
+            return this[TEMPLATE_INJECTION_ERROR](
```

Comment:
It will return undefined ([ref](https://github.com/fiverr/i18n.js/pull/87/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R44)). I found this logic in [the DR](https://docs.google.com/presentation/d/1_LVxPlN_kGn-Fx3mhEM7ALVWgn-2ta0-1NEIalmQvdE/edit#slide=id.g245b7d7db16_0_5) and original PR. IMHO it might be useful for some complex logic in consumer.

Reply:
Let's rethink it plz. The consumer should be really as dumb as possible since it's a global one and not something is controlled by the developer who applies the translation.